File: openapi-test-kotlin-server-spring/testapi/api/PetsApi.kt
package testapi.api

import com.fasterxml.jackson.databind.JsonNode
import java.lang.Void
import java.util.Optional
import kotlin.collections.List
import testapi.model.Pet
import testapi.model.PetCreate

sealed interface PetsApi {
  /** Create a pet */
  fun createPet(body: PetCreate): CreatePetResponse

  /** Delete a pet */
  fun deletePet(
    /** The pet ID */
    petId: String
  ): DeletePetResponse

  /** Get a pet by ID */
  fun getPet(
    /** The pet ID */
    petId: String
  ): GetPetResponse

  /** Get pet photo */
  fun getPetPhoto(
    /** The pet ID */
    petId: String
  ): Void

  /** List all pets */
  fun listPets(
    /** Maximum number of pets to return */
    limit: Optional<Integer>,
    /** Filter by status */
    status: Optional<String>
  ): List<Pet>

  /** Upload a pet photo */
  fun uploadPetPhoto(
    /** The pet ID */
    petId: String,
    /** Optional caption for the photo */
    caption: String,
    /** The photo file to upload */
    file: Array<Byte>
  ): JsonNode
}

File: openapi-test-scala-server-spring-client-jdk/testapi/api/PetsApi.scala
package testapi.api

import io.circe.Json
import java.lang.Exception
import java.util.Optional
import testapi.model.Error
import testapi.model.Pet
import testapi.model.PetCreate
import testapi.model.PetId

trait PetsApi {
  /** Create a pet */
  @throws[Exception]
  def createPet(body: PetCreate): Response201400[Pet, Error]

  /** Delete a pet */
  @throws[Exception]
  def deletePet(
    /** The pet ID */
    petId: PetId
  ): Unit

  /** Get a pet by ID */
  @throws[Exception]
  def getPet(
    /** The pet ID */
    petId: PetId
  ): Response200404[Pet, Error]

  /** Get pet photo */
  @throws[Exception]
  def getPetPhoto(
    /** The pet ID */
    petId: PetId
  ): Unit

  /** List all pets */
  @throws[Exception]
  def listPets(
    /** Maximum number of pets to return */
    limit: Optional[Integer],
    /** Filter by status */
    status: Optional[String]
  ): java.util.List[Pet]

  /** Upload a pet photo */
  @throws[Exception]
  def uploadPetPhoto(
    /** The pet ID */
    petId: PetId,
    /** Optional caption for the photo */
    caption: String,
    /** The photo file to upload */
    file: Array[Byte]
  ): Json
}